react/features/base/profile/middleware.js

```diff
@@ -18,11 +19,28 @@ MiddlewareRegistry.register(store => next => action => {
     switch (action.type) {
     case PROFILE_UPDATED:
         _updateLocalParticipant(store);
+        _maybeUpdateStartAudioOnly(store, action);
     }
 
     return result;
 });
 
+/**
+ * Updates startAudioOnly flag if it's updated in the profile.
+ *
+ * @private
+ * @param {Store} store - The redux store.
+ * @param {Object} action - The redux action.
+ * @returns {void}
+ */
+function _maybeUpdateStartAudioOnly(store, action) {
+    const { profile } = action;
+
+    if (typeof profile.startAudioOnly === 'boolean') {
```

Comment:
Is there a time when startAudioOnly is initialized as undefined instead of false? Or is there a time when it's not a boolean?

Reply:
No, it's just that there can be a profile update call when the startAudioOnly is not modified. And in those cases invoking the related dispatch can have bad side effects (destroying/creating tracks)